climpred_skeleton/comparison.py

```diff
@@ -0,0 +1,95 @@
+from typing import Union
+
+import pkg_resources
+import xarray as xr
+import yaml
+
+from .core import Verification
+
+
+class Comparison(Verification):
+    """First step in pipeline. Broadcast your prediction and observations
+    in a certain manner.
+
+    Base Comparison class hints at which attributes are expected for each
+    sub comparison class. It also stores the factory for internal use, to
+    return the appropriate subfunction.
+    """
+
+    _comparison_type: str
+    _hindcast_comparison: bool
+    _probabilistic_comparison: bool
+
+    def __init__(self, prediction: xr.Dataset, observation: xr.Dataset) -> None:
+        super().__init__(prediction, observation)
+
+    def factory(self, type) -> 'Comparison':
+        try:
+            return comparison_dict[type](self._prediction, self._observation)
+        except KeyError:
+            raise ValueError(
+                f'{type} not valid keyword from {list(comparison_dict.keys())}'
+            )
+
+
+class e2o(Comparison):
+    """Example subclass for Comparison.
+
+    Stores useful information (e.g. whether this is a probabilistic comparison)
+    and houses main `broadcast` function, returning the modified prediction
+    and observation objects.
+    """
+
+    def __init__(self, prediction: xr.Dataset, observation: xr.Dataset) -> None:
+        super().__init__(prediction, observation)
+        self._comparison_type = 'e2o'
```

Comment:
Don't know if it's better to explicitly hardcode the name or dynamically retrieve e.g. `self.__class__.__name__.lower()`
probably fine to hardcode the name for readability